test/integration/iterate_by_column_test.rb

```diff
@@ -0,0 +1,118 @@
+  def test_specifying_pagination_column_by_per_table_succeeds
+    pagination_key_column = 'some_column'
+    db_name = DEFAULT_DB
+    table_name = DEFAULT_TABLE
+
+    create_tables(db_name, table_name, "(#{pagination_key_column} bigint(20), data BLOB)")
+    seed_dynamically(DbHelper.full_table_name(db_name, table_name), {
+      pagination_key_column => lambda {|i| i},
+      data: lambda {|_| "'some data'"}
+    }, amount_rows: 10)
+
+    assert_ghostferry_run(per_table: {
+      db_name => {
+        table_name => pagination_key_column,
+      },
+    })
+    assert_source_target_eql_data(DbHelper.full_table_name(db_name, table_name))
+  end
+
+  def test_specifying_pagination_column_by_fallback_column_succeeds
+    pagination_key_column = 'some_column'
+    db_name = DEFAULT_DB
+    table_name = DEFAULT_TABLE
+
+    create_tables(db_name, table_name, "(#{pagination_key_column} bigint(20), data BLOB)")
+    seed_dynamically(DbHelper.full_table_name(db_name, table_name), {
+      pagination_key_column => lambda {|i| i},
+      data: lambda {|_| "'some data'"}
+    }, amount_rows: 10)
+
+    assert_ghostferry_run(fallback_column: pagination_key_column)
+    assert_source_target_eql_data(DbHelper.full_table_name(db_name, table_name))
+  end
+
+  def test_specifying_pagination_column_by_per_table_and_fallback_column_succeeds
+    pagination_key_column_1 = 'some_column'
+    pagination_key_column_2 = 'other_column'
+
+    db_name = DEFAULT_DB
+    table_name_1 = DEFAULT_TABLE
+    table_name_2 = "other_table"
+
+    create_tables(db_name, table_name_2, "(#{pagination_key_column_2} bigint(20), data BLOB)")
+    # Following creates composite index: other_id, some_column
+    create_tables(db_name, table_name_1, "(#{pagination_key_column_1} bigint(20), other_id bigint(20), data BLOB, primary key(other_id, #{pagination_key_column_1}))")
+
+
+    seed_dynamically(DbHelper.full_table_name(db_name, table_name_1), {
+      pagination_key_column_1 => lambda {|i| i},
+      other_id: lambda {|i| i},
+      data: lambda {|_| "'some data'"}
+    }, amount_rows: 10)
+
+    seed_dynamically(DbHelper.full_table_name(db_name, table_name_2), {
+      pagination_key_column_2 => lambda {|i| i},
+      data: lambda {|_| "'some data'"}
+    }, amount_rows: 10)
+
+    assert_ghostferry_run(per_table: {
+      db_name => {
+        table_name_1 => pagination_key_column_1,
+      },
+    }, fallback_column: pagination_key_column_2)
+
+    assert_source_target_eql_data(DbHelper.full_table_name(db_name, table_name_1))
+    assert_source_target_eql_data(DbHelper.full_table_name(db_name, table_name_2))
+  end
+
+  private
+  def create_tables(db_name, table_name, create_query)
+    [source_db, target_db].each do |db|
+      db.query("CREATE DATABASE IF NOT EXISTS #{db_name}")
+      db.query("CREATE TABLE IF NOT EXISTS #{DbHelper.full_table_name(db_name, table_name)} #{create_query}")
+    end
+  end
+
+  def seed_dynamically(full_table_name, columns, amount_rows:)
```

Comment:
If this is deemed to make sense, then I will promote as helper function like: https://github.com/Shopify/ghostferry/blob/b5a11f7f8b8ae58d8811653e3a310e0bdcd2e87f/test/helpers/db_helper.rb#L88

Reply:
We can leave it here for now until we need it somewhere else.